packages/adapters/txservice/src/aggregator.ts

```diff
@@ -678,25 +679,71 @@ export class RpcProviderAggregator {
     if (needsSigner) {
       this.checkSigner();
     }
+
     const errors: NxtpError[] = [];
-    const shuffledProviders = this.shuffleSyncedProviders();
-    for (const provider of shuffledProviders) {
-      try {
-        return await method(provider);
-      } catch (e: unknown) {
-        // TODO: With the addition of SyncProvider, this parse call may be entirely redundant. Won't add any compute,
-        // however, as it will return instantly if the error is already a NxtpError.
-        const error = parseError(e);
-        if (error.type === ServerError.type || error.type === RpcError.type || error.type === StallTimeout.type) {
-          // If the method threw a StallTimeout, RpcError, or ServerError, that indicates a problem with the provider and not
-          // the call - so we'll retry the call with a different provider (if available).
-          errors.push(error);
-        } else {
-          // e.g. a TransactionReverted, TransactionReplaced, etc.
-          throw error;
+    const handleError = (e: unknown) => {
+      // TODO: With the addition of SyncProvider, this parse call may be entirely redundant. Won't add any compute,
+      // however, as it will return instantly if the error is already a NxtpError.
+      const error = parseError(e);
+      if (error.type === ServerError.type || error.type === RpcError.type || error.type === StallTimeout.type) {
+        // If the method threw a StallTimeout, RpcError, or ServerError, that indicates a problem with the provider and not
+        // the call - so we'll retry the call with a different provider (if available).
+        errors.push(error);
+      } else {
+        // e.g. a TransactionReverted, TransactionReplaced, etc.
+        throw error;
+      }
+    };
+
+    const quorum = this.config.quorum ?? 1;
+    if (quorum > 1) {
+      // Consult ALL providers.
+      const results: (T | undefined)[] = await Promise.all(
+        this.providers.map(async (provider) => {
+          try {
+            return await method(provider);
+          } catch (e: unknown) {
+            handleError(e);
+            return undefined;
+          }
+        }),
+      );
+      // Filter out undefined results.
+      // NOTE: If there aren't any defined results, we'll proceed out of this code block and throw the
+      // RpcError at the end of this method.
+      const filteredResults: T[] = results.filter((item) => item !== undefined) as T[];
+      if (filteredResults.length > 0) {
+        // Pick the most common answer.
+        let counts: Map<T, number> = new Map();
+        counts = filteredResults.reduce((counts, item) => {
```

Comment:
`item` may not be a string since its part of the returned provider value

Reply:
and im not sure if this keying will work to identify the same returned object

Reply:
With `Map` you can key using a generic type afaik... wouldn't I be getting compile issues if this were true? When I highlight `item` at any point, it seems to think the generic type `T` is maintained

Reply:
TBH I'm pretty sure it's `string` in all cases though

Reply:
provider responses? i dont think those are always strings. but im saying im not sure if its going to be indexed as the same value if its an object, not that it isn't technically supported. not sure though